src/frontend/apps/markdown/MdxRenderingOptionsSelector/index.spec.tsx

```diff
@@ -0,0 +1,103 @@
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import React from 'react';
+
+import { wrapInIntlProvider } from 'utils/tests/intl';
+
+import MdxRenderingOptionsSelector from '.';
+
+describe('<MdxRenderingOptionsSelector />', () => {
+  it('send options change', async () => {
+    window.scrollTo = jest.fn(); // required to test, see grommet
+
+    const setRenderingOptions = jest.fn();
+
+    const { rerender } = render(
+      wrapInIntlProvider(
+        <MdxRenderingOptionsSelector
+          renderingOptions={{ useMdx: false }}
+          setRenderingOptions={setRenderingOptions}
+        />,
+      ),
+    );
+
+    const dropButton = screen.getByRole('button', { name: 'Settings' });
+    userEvent.click(dropButton); // open the menu
+
+    // Toggle MDX
+    const toggleMdxButton = screen.getByTestId('mdx-options-toggle-mdx');
+
+    expect(screen.getByText('MDX disabled')).toBeInTheDocument();
+    expect(screen.getByText('Mathjax disabled')).toBeInTheDocument();
+
+    userEvent.click(toggleMdxButton);
+
+    expect(setRenderingOptions).toHaveBeenCalledTimes(1);
+    expect(setRenderingOptions).toHaveBeenCalledWith({ useMdx: true });
+
+    rerender(
+      wrapInIntlProvider(
+        <MdxRenderingOptionsSelector
+          renderingOptions={{ useMdx: true }}
+          setRenderingOptions={setRenderingOptions}
+        />,
+      ),
+    );
+
+    expect(screen.getByText('MDX enabled')).toBeInTheDocument();
+
+    // Toggle Mathjax
+    const toggleMathjaxButton = screen.getByTestId(
```

Comment:
Most of the `getByTestId` can be replaced by `getByRole('checkbox', { name: '...'})` in this file. I highly suggest you to use `getByTestId` if there is no other solution. Usin `getByRole` here you will also check if the label is the one expected.